json-rpc/tests/integration_test.rs

```diff
@@ -719,8 +719,8 @@ fn create_test_cases() -> Vec<Test> {
                             "XUS"
                         ],
                         "arguments": [
-                            "{ADDRESS: 000000000000000000000000000000DD}",
-                            "{U64: 100}",
+                            "[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 221]",
```

Comment:
yeah this seams like it could be a breaking change, but i'd need to look at it a little closer

Reply:
So i took a closer look and while I'm still not convinced this isn't a breaking change, I think this is an improper way to display this data. If we want to send back the bytes, then lets do what do everywhere and send hex encoded bcs bytes instead of unparsable, array of ints encoded in a string.

Reply:
> send hex encoded bcs bytes instead of unparsable, array of ints encoded in a string.

I agree with this, and this will make the display a little bit prettier. But I don't think we can get back the
`{ADDRESS: 000000000000000000000000000000DD}` or `{U64: 100}` because the type information no longer exists.

Reply:
I'm wondering if we want to build a mapping of `script_function_name => script_abi` in the building of the DF, and use that here to determine the types of these arguments to then deserialize them here?  My main concern beyond the backwards compatibility is that from a client experience point of view this might be putting a lot of work on them to understand the arguments that they don't have with normal transaction scripts.

Reply:
I'm not expecting you to get back the type info here, the arguments are being changed to BCS bytes, so lets ship hex encoded BCS bytes here. The bigger question is, does it make more sense to have a new "arguments_bcs" field that these are rendered to instead of the existing "arguments" field.

Reply:
I think I'd probably be in favor of adding the `arguments_bcs` since the types would work out a bit odd if we tried to reuse `arguments` since `arguments: Option<Vec<String>>` and we really want an `Option<Vec<BytesView>>` for these which feels a bit weird to me. 

Additionally, since we're already adding other fields for script functions, it seems more natural to add the new field IMO. Interested in what @mengxu-fb thinks about `arguments_bcs` vs `arguments` as well.

Reply:
First let me clarify whether the following diff captures the changes being proposed:

```diff
// json-rpc/types/src/views.rs
pub struct ScriptView {
    ...
    // script arguments, converted into string with type information.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub arguments: Option<Vec<String>>,

+   // script function arguments, converted into hex encoded BCS bytes
+   #[serde(skip_serializing_if = "Option::is_none")]
+   pub arguments_bcs: Option<Vec<BytesView>>,
    ...
}
```

I am OK with the additional `arguments_bcs` field, and in theory, only one of `arguments` or `arguments_bcs` will be set and the other should be `None`.

But I don't quite see the reason why we have to make a change. In my understanding of the code, this information is mainly provided to the user for information (and maybe debugging?) only, and I don't think they can reconstruct anything from the `String`, being it formatted map view like `{U64: 100}` or just raw bytes.

Reply:
> I'm wondering if we want to build a mapping of `script_function_name => script_abi` in the building of the DF, and use that here to determine the types of these arguments to then deserialize them here?

I like this idea more, if I have to choose between
1) shipping raw hex bytes in `arguments_bcs`, vs
2) deserialize them with the ABI info and ship formatted string in `arguments`.

Reply:
@mengxu-fb Yes that diff does capture what I was suggesting, but it sounds like you have another approach you're looking into, I don't have too much of an opinion on which you choose, just that you should take into consideration how you'd render the type information in the future when these types could be more complex.

Reply:
Sitting on the thought for a night, I think we can have both `arguments_bcs` (which gives the raw hex bytes) and `arguments` (which gives typed value if deserialization is successful) for ScriptFunction. As a starter, I can add the `arguments_bcs` first, and think about the `arguments` later.

Reply:
`arguments_bcs` added in commit bfc8d88